test/test_intervals.py

```diff
@@ -230,6 +231,98 @@ def testOverlappingIntervalsThatCoverEverything(self):
             list(ri.walk()))
 
     # The following tests have the same setup as above, but they test the
+    # baseCoverage() method.
+
+    def testEmptyBaseCoverageOnZeroLengthSequence(self):
+        """
+        When no intervals are added, baseCoverage should return an empty
+        Counter.
+        """
+        ri = ReadIntervals(0)
+        self.assertEqual(Counter(), ri.baseCoverage())
+
+    def testEmptyBaseCoverage(self):
+        """
+        When no intervals are added, baseCoverage should return an empty
+        Counter.
+        """
+        ri = ReadIntervals(100)
+        self.assertEqual(Counter(), ri.baseCoverage())
+
+    def testOneIntervalExactCoveringBaseCoverage(self):
+        """
+        If there is a single interval that spans the whole hit exactly,
+        baseCoverage should return the correct result.
+        """
+        ri = ReadIntervals(10)
+        ri.add(0, 10)
+        c = Counter([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        self.assertEqual(c, ri.baseCoverage())
+
+    def testOneIntervalCoveringAllExtendingLeftBaseCoverage(self):
+        """
+        If there is a single interval that spans the whole hit, including
+        going negative to the left, baseCoverage should return the correct
+        result.
+        """
+        ri = ReadIntervals(10)
+        ri.add(-2, 10)
+        c = Counter([-2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        self.assertEqual(c, ri.baseCoverage())
```

Comment:
I don't think this make sense. Look at what the `coverage` method does. You shouldn't return anything for negative indices - they don't exist. The method is returning information about the subject indices, not about the reads (which may overlap to the left/right).